application/src/main/kotlin/application/BackwardCompatibilityCheckCommand.kt

```diff
@@ -0,0 +1,327 @@
+package application
+
+import application.BackwardCompatibilityCheckCommand.CompatibilityResult.*
+import io.specmatic.conversions.OpenApiSpecification
+import io.specmatic.core.*
+import io.specmatic.core.git.GitCommand
+import io.specmatic.core.git.SystemGit
+import io.specmatic.core.log.logger
+import io.specmatic.core.utilities.exitWithMessage
+import io.specmatic.stub.isOpenAPI
+import org.springframework.stereotype.Component
+import picocli.CommandLine.Command
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.concurrent.Callable
+import java.util.regex.Pattern
+import kotlin.io.path.extension
+import kotlin.io.path.pathString
+import kotlin.system.exitProcess
+
+const val ONE_INDENT = "  "
+const val TWO_INDENTS = "${ONE_INDENT}${ONE_INDENT}"
+
+@Component
+@Command(
+    name = "backwardCompatibilityCheck",
+    mixinStandardHelpOptions = true,
+    description = [
+        "Checks backward compatibility of a directory across the current HEAD and the main branch",
+        "DEPRECATED: This command will be removed in the next major release. Use 'backward-compatibility-check' command instead."
```

Comment:
Let's put a date or the release number when we will be completely deprecating/removing this.